src/app/App.js

```diff
@@ -50,40 +50,27 @@ export default class App {
   }
 
   async getImplementation(contractName) {
-    const directory = this.currentDirectory()
-    return directory.getImplementation(contractName)
+    return this.package.getImplementation(this.version, contractName)
   }
 
   async getProxyImplementation(proxyAddress) {
     return this._app.getProxyImplementation(proxyAddress, this.txParams)
   }
 
   async setImplementation(contractClass, contractName) {
-    log.info(`Setting implementation of ${contractName} in directory...`)
-    const implementation = await contractClass.new(this.txParams)
-    const directory = this.currentDirectory()
-    await directory.setImplementation(contractName, implementation.address, this.txParams)
-    log.info(` Implementation set: ${implementation.address}`)
-    return implementation
+    return this.package.setImplementation(this.version, contractClass, contractName)
   }
 
   async setStdlib(stdlibAddress = 0x0) {
-    log.info(`Setting stdlib ${stdlibAddress}...`)
-    await this.currentDirectory().setStdlib(stdlibAddress, this.txParams)
-    return stdlibAddress
+    return this.currentDirectory().setStdlib(stdlibAddress)
```

Comment:
Doesn't this return a tx receipt? Shouldn't we be hiding that?

Reply:
Ignore this. I was wrong.

Reply:
No, the directory is a wrapped `AppDirectory` here, it will return the address directly